src/averageAges.js

```diff
@@ -55,7 +71,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => {
+    return onlyWithSon
+      ? person.sex === 'm' && hasMother(people, person.mother)
+      : hasMother(people, person.mother);
+  });
+
+  const ageDiffrences = children.map(child =>
+    child.born - motherBorn(people, child.mother));
+
+  const averageDiff = ageDiffrences.reduce((sum, age) => sum + age);
```

Comment:
I recommend to always give initial value in reduce, as there will be error for an empty array, if you don't provide the initial value